metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -3457,33 +3493,88 @@ def _create_workbook_properties(
         return {"permissions": json.dumps(groups)} if len(groups) > 0 else None
 
     def ingest_tableau_site(self):
+        self.report.report_ingestion_stage_start(
+            f"Ingesting Tableau Site: {self.site_id} {self.site_content_url}"
+        )
+
         # Initialise the dictionary to later look-up for chart and dashboard stat
         if self.config.extract_usage_stats:
-            self._populate_usage_stat_registry()
+            with PerfTimer() as timer:
+                self._populate_usage_stat_registry()
+                self.report.extract_usage_stats_timer[self.site_id] = round(
+                    timer.elapsed_seconds(), 2
+                )
 
         if self.config.permission_ingestion:
-            self._fetch_groups()
+            with PerfTimer() as timer:
+                self._fetch_groups()
+                self.report.fetch_groups_timer[self.site_id] = round(
```

Comment:
might be nicer to have site names in here instead of site ids, assuming they're unique